drivers/source/UARTSerial.cpp

```diff
@@ -351,8 +337,15 @@ void UARTSerial::tx_irq(void)
 /* These are all called from critical section */
 void UARTSerial::enable_rx_irq()
 {
-    SerialBase::attach(callback(this, &UARTSerial::rx_irq), RxIrq);
-    _rx_irq_enabled = true;
+    core_util_critical_section_enter();
```

Comment:
Comment above at line 337 is no longer true. Needs moving down.

It's a bit odd having `enable_rx_irq` do the critical section and `disable_rx_irq` not. And indeed them not being symmetrical. 

Not totally against it, but maybe at least change the name of `enable_rx_irq` to indicate it's more complex, and not just the opposite.

Reply:
Yes I kind of tried to keep that symmetry, which is confusing in this case.

What about renaming `enable_rx_irq` to `update_rx_irq`?

Reply:
Did you miss / misinterpret my note about the "These are all called from critical section" comment?

Reply:
Yes, thought you were talking about the "only write to hardware in one place" comment. Pushed an update.